nav2_simple_commander/launch/test_error_codes_launch.py

```diff
@@ -0,0 +1,45 @@
+# Copyright (c) 2022 Joshua Wallace
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+
+from ament_index_python.packages import get_package_share_directory
+
+from launch import LaunchDescription
+from launch.actions import ExecuteProcess, IncludeLaunchDescription
+from launch.launch_description_sources import PythonLaunchDescriptionSource
+from launch_ros.actions import Node
+
+
+def generate_launch_description():
+    # warehouse_dir = get_package_share_directory('aws_robomaker_small_warehouse_world')
+    nav2_bringup_dir = get_package_share_directory('nav2_bringup')
+
+    # start navigation
+    bringup_cmd = IncludeLaunchDescription(
+        PythonLaunchDescriptionSource(
+            os.path.join(nav2_bringup_dir, 'launch', 'tb3_simulation_launch.py')),
+        launch_arguments={'headless': 'True',
+                          'use_rviz': 'False'}.items())
+
+    test_error_code = Node(
+        package='nav2_simple_commander',
+        executable='test_error_codes',
+        emulate_tty=True,
+        output='screen')
+
+    ld = LaunchDescription()
```

Comment:
Look at the system tests for Nav2 for examples of how to do a testing script with launch. There's a testing API that will launch some nodes + also launch a "testing script". It works such that once the testing script exits, it kills everything since the test is over